ADAL/src/public/ADAuthenticationContext.h

```diff
@@ -520,6 +523,26 @@ typedef enum
                               userId:(nonnull NSString *)userId
                      completionBlock:(nonnull ADAuthenticationCallback)completionBlock;
 
+/*! Follows the OAuth2 protocol (RFC 6749). The function accepts claims challenge returned from middle tier service, which will be sent to authorization endpoint. ADAL will ignore cache and will not attempt
+ to silently acquire token or return access token from cache. It will get the token through webview.
+ @param resource The resource for whom token is needed.
+ @param clientId The client identifier
+ @param redirectUri The redirect URI according to OAuth2 protocol
+ @param promptBehavior Controls if any credentials UI will be shown.
+ @param userId An ADUserIdentifier object describing the user being authenticated
+ @param queryParams The extra query parameters will be appended to the HTTP request to the authorization endpoint. This parameter can be nil. It should be URL-encoded.
+ @param claims The claims parameter that needs to be sent to authorization endpoint. It should be URL-encoded.
+ @param completionBlock the block to execute upon completion. You can use embedded block, e.g. "^(ADAuthenticationResult res){ <your logic here> }"
+ */
+- (void)acquireTokenInteractiveWithResource:(nonnull NSString *)resource
```

Comment:
why was this added?

Reply:
This method was added to improve performance for one of our scenarios. Our app already tries to silently get an access token to a resource from another code path and know that consent is required after making 2 failed requests due to FOCI. If we then try to use the "133" API passing in AD_PROMPT_AUTO, that method will attempt 2 more tries to silently get the access token (4 identical silent requests). AD_PROMPT_ALWAYS skips the silent request, but it appends "prompt=login" to the web view request forcing user to enter password. The only difference between these two API is that this one sets skip cache so that the web view will always be used.

Reply:
Sounds good, resolving this one.